tracer/src/Datadog.Trace/Configuration/ServiceNames.cs

```diff
@@ -49,7 +59,13 @@ public void SetServiceNameMappings(IEnumerable<KeyValuePair<string, string>> map
         {
             lock (_lock)
             {
-                _mappings = mappings.ToDictionary(x => x.Key, x => x.Value);
+                var mappingsDictionary = new Dictionary<string, string>();
```

Comment:
Can we initialize this `Dictionary<>` with the correct size from `mappings`?

Reply:
nope because it will force an enumeration.